sdk/tests/js/glsl-conformance-test.js

```diff
@@ -162,8 +162,10 @@ function runOneTest(gl, info) {
       }
     }
     // Save the shaders so we test shared shader.
-    if (vShader) {
-      vShaderDB[vSource] = vShader;
+    if (compiledVShader) {
+      vShaderDB[vSource] = compiledVShader;
+    } else {
+      vShader = null;
```

Comment:
Use compiledVShader wherever vShader is used below instead. The meaning of variables should avoid changing over the course of a function.

Reply:
compiledVShader and compiledFShader are scoped within their blocks and are inaccessible to the code below. Feel free to refactor further as long as the tests continue to pass.